submissions/Chikus/port-sniffer/sniffer.js

```diff
@@ -0,0 +1,69 @@
+const Net = require('net');
+var host = '';
+var lower_port = 0;
+var higher_port = 65535;
+var port_list = [];
+var timeout = 300;
+
+function check(host, port, callback) {
+ var socket = Net.createConnection(port, host);
+ var timer  = setTimeout(function () {
+  socket.destroy();
+  callback(false);
+}, timeout);
+ socket.once('connect', function () {
+  clearTimeout(timer);
+  socket.destroy();
+  process.stdout.write('.');
+  port_list.push(port);
+  callback(true);
+ });
+ socket.on('error', function () {
+  clearTimeout(timer);
+  callback(false);
+ });
+}
+
+function run_ports(){
+  check(host,lower_port, function next_check(result) {
+    if (lower_port == higher_port) {
```

Comment:
Its better to always use `===` to prevent assertion bugs

Reply:
Done, Thanks!

Reply:
Done!